(GH-1376) Update bundled content to use $targets parameter

beechtom opened the pull request:
This updates the `canary`, `puppetdb_fact`, and `aggregate` modules to use a `$targets` parameter in favor of `$nodes`. It also updates relevant documentation to reflect this change and drops all references to `nodes` in favor of `targets`.

Closes #1376

beechtom commented:
[This page](https://github.com/puppetlabs/bolt/pull/1410/files#diff-a9dd11a262f58840576fabf6f1c2e589) has all references to `nodes` replaced with `targets`. Doing this replacement is pretty quick, so do we want to go ahead and do the same for the rest of the docs and hands-on lab?

donoghuc commented:
I think doing the `nodes -> targets` replacement everywhere you call out is reasonable. It will be somewhat tedious to review but i think a big push to get it done will be good and worth the risk of accidentally breaking some examples for the sake of clarity over the preference for targets.

puppetcla commented:
CLA signed by all contributors.

donoghuc commented on modules/aggregate/lib/puppet/functions/aggregate/targets.rb:
```diff
@@ -1,14 +1,14 @@
 # returning that value for the key.
-Puppet::Functions.create_function(:'aggregate::nodes') do
-  dispatch :aggregate_nodes do
+Puppet::Functions.create_function(:'aggregate::targets') do
```
I don't think it is worth this breaking change to rename the function. But I might be wrong...

nicklewis replied:
Renaming the param is also a breaking change. I'm not sure how we want to handle that

beechtom replied:
Fwiw, our docs do say that versions of bundled modules may have breaking changes.

lucywyman replied:
We could have both available, only document `aggregate::targets`, and remove `nodes` in 2.0? just removing `aggregate::nodes` in a y release feels...unnecessarily unfriendly.

lucywyman commented on bolt-modules/boltlib/lib/puppet/functions/catch_errors.rb:
```diff
@@ -12,14 +12,14 @@
   #         otherwise the result will be returned
```
Should this also update `node` to `target` in the the spec directory here? Just while we're here:
```
lucy ~/githubs/bolt ((cf68954...)) $ ag -rl "node" bolt-modules/*
bolt-modules/boltlib/spec/functions/puppetdb_fact.rb
bolt-modules/boltlib/spec/functions/run_plan_spec.rb
bolt-modules/boltlib/spec/functions/run_command_spec.rb
bolt-modules/boltlib/spec/functions/puppetdb_query.rb
bolt-modules/boltlib/spec/functions/upload_file_spec.rb
bolt-modules/boltlib/spec/functions/run_task_spec.rb
bolt-modules/boltlib/spec/functions/apply_prep_spec.rb
bolt-modules/boltlib/spec/functions/run_script_spec.rb
bolt-modules/boltlib/spec/functions/get_resources_spec.rb
```

lucywyman replied:
Also update in `documentation/bolt_configuration_options.md`, `doc/index.html, and `doc/file.README.html`

lucywyman commented on docs/_lab/03-running-commands/index.md:
```diff
@@ -4,98 +4,98 @@ difficulty: Basic
-Successful on 1 node: node1
-Ran on 1 node in 0.42 seconds
+Successful on 1 target: target1
+Ran on 1 target in 0.42 seconds
```
I don't think we've changed this output yet...https://github.com/puppetlabs/bolt/blob/master/lib/bolt/outputter/human.rb#L170 Should that be part of this change?

lucywyman commented on bolt-modules/boltlib/lib/puppet/functions/run_plan.rb:
```diff
@@ -11,21 +11,21 @@
   # @return [PlanResult] The result of running the plan. Undef if plan does not explicitly return results.
   # @example Run a plan
-  #   run_plan('canary', $nodes, 'command' => 'false')
+  #   run_plan('canary', $targets, 'command' => 'false')
```
This is going to fail for plans that have a `TargetSpec $targets` parameter since this explicitly passes the `nodes` argument - I think this is probably a bug from #1411  .

lucywyman commented:
Needs a rebase. It also looks like a few more 'nodes' may have snuck in?
```
lucy ~/githubs/bolt ((31a927f...)) $ ag -r "nodes" bolt-modules/* --ignore vendor*
bolt-modules/boltlib/spec/functions/remove_from_group_spec.rb
19:        'nodes' => [target1],
22:            'nodes' => [target1, target2] }

bolt-modules/boltlib/spec/functions/run_plan_spec.rb
147:  context 'with a plan with both a $nodes and $targets parameter' do
159:        is_expected.to run.with_params('test::run_me_nodes_and_targets', 'target1')
165:      it 'specifies the $nodes parameter using the second positional argument' do
166:        is_expected.to run.with_params('test::run_me_nodes_and_targets', 'target1', 'targets' => 'target2')

bolt-modules/boltlib/spec/functions/get_resources_spec.rb
67:        Bolt::RunFailure, "Plan aborted: run_task 'query_resources_task' failed on #{targets.count} nodes"

bolt-modules/boltlib/spec/functions/apply_prep_spec.rb
110:        Bolt::RunFailure, "Plan aborted: run_task 'puppet_agent::version' failed on 2 nodes"
124:        Bolt::RunFailure, "Plan aborted: run_task 'custom_facts_task' failed on #{targets.count} nodes"
132:          'nodes' => [{
```

beechtom commented:
The first file has `nodes` since it has tests written for inventory v1. We can rip those out with 2.0. The second file has tests for the `run_plan` function to make sure the second positional argument is mapped to either `$nodes` or `$targets` correctly.

The last two files are matches for the `Bolt::RunError` error message. I've updated that to show `targets` instead (and updated it to only write `target` if there was a single target).

lucywyman commented on CHANGELOG.md:
```diff
@@ -14,6 +14,10 @@
 
+* **Bundled content updated to use `$targets` parameter** ([#1376](https://github.com/puppetlabs/bolt/issues/1376))
+
+  Plans that are part of the `canary`, `puppetdb_fact`, and `aggregate` modules have been updated to use a `$targets` parameter in favor of `$nodes`. The `aggregate::nodes` plan still uses a `$nodes` parameter, but the module now includes a `aggregate::targets` plan that uses a `$targets` parameter.
```
```suggestion
  Plans that are part of the `canary`, `puppetdb_fact`, and `aggregate` modules have been updated to use a `$targets` parameter instead of `$nodes`. The `aggregate::nodes` plan still uses a `$nodes` parameter, but the module now includes a `aggregate::targets` plan that uses a `$targets` parameter.
```
